osquery/tables/system/darwin/gatekeeper.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/sql/sqlite_util.h"
+
+namespace fs = boost::filesystem;
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+const std::string kGkeStatusPath{
+    "/var/db/SystemPolicy-prefs.plist"};
+
+const std::string kGkeBundlePath{
+    "/var/db/gke.bundle/Contents/version.plist"};
+
+const std::string kGkeOpaquePath{
+    "/var/db/gkopaque.bundle/Contents/version.plist"};
+
+const std::string kPolicyDb = "/var/db/SystemPolicy";
+
+bool isGateKeeperDevIdEnabled() {
+  sqlite3* db = nullptr;
+  auto rc = sqlite3_open_v2(
+      kPolicyDb.c_str(),
+      &db,
+      (SQLITE_OPEN_READONLY | SQLITE_OPEN_PRIVATECACHE | SQLITE_OPEN_NOMUTEX),
+      nullptr);
+  if (rc != SQLITE_OK || db == nullptr) {
+    VLOG(1) << "Cannot open Gatekeeper DB: " << rc << " "
+            << getStringForSQLiteReturnCode(rc);
+    if (db != nullptr) {
+      free(db);
+    }
+  }
+
+  std::string query = "SELECT disabled FROM authority WHERE label = 'Developer ID'";
+  sqlite3_stmt* stmt = nullptr;
+  rc = sqlite3_prepare_v2(db, query.c_str(), -1, &stmt, nullptr);
+
+  while ((rc = sqlite3_step(stmt)) == SQLITE_ROW) {
+    int value = sqlite3_column_int(stmt, 0);
+    if (value == 1) {
+      // Clean up.
+      sqlite3_finalize(stmt);
+      free(db);
+      return false; // return false if any rows say "disabled"
+    }
+  }
+  sqlite3_finalize(stmt);
+  free(db);
+  return true;
+}
+
+QueryData genGateKeeper(QueryContext& context) {
+  Row r;
+
+  auto gke_status = SQL::selectAllFrom("plist", "path", EQUALS, kGkeStatusPath);
+
+  if (gke_status.empty()) {
+    r["assessments_enabled"] = INTEGER(0);
+  }
+
+  for (const auto& row : gke_status) {
+    if (row.at("key") == "enabled" && row.at("value") == "yes" ) {
```

Comment:
nit: As you're opening an external db, you might wanna check that these keys exist before querying their values to prevent throws. Maybe just
```
if (row.find("key") == row.end() || row.find("value") == row.end() {
  continue;
}
```